Build appears broken on DragonFlyBSD 5.4

```
/home/worker/test-rust-libc.sh
 in dir /home/worker/buildbot/dfbsd54-build/build (timeout 1200 secs)
 watching logfiles {}
 argv: [b'/home/worker/test-rust-libc.sh']
 environment:
  BLOCKSIZE=K
  EDITOR=vi
  ENV=/home/worker/.shrc
  HOME=/home/worker
  LOGNAME=worker
  MAIL=/var/mail/worker
  OLDPWD=/home/worker
  PAGER=less
  PATH=/sbin:/bin:/usr/sbin:/usr/bin:/usr/games:/usr/local/sbin:/usr/local/bin:/usr/pkg/sbin:/usr/pkg/bin:/home/worker/bin
  PWD=/home/worker/buildbot/dfbsd54-build/build
  SHELL=/usr/local/bin/bash
  SHLVL=1
  SUDO_COMMAND=/usr/local/bin/bash
  SUDO_GID=1001
  SUDO_UID=1001
  TERM=xterm-256color
  USER=worker
  _=/usr/local/bin/buildbot-worker-3.6
 using PTY: False
DragonFly bb-dfbsd54. 5.4-RELEASE DragonFly v5.4.1-RELEASE #2: Sat Dec 22 17:43:09 EST 2018     root@www.shiningsilence.com:/usr/obj/home/justin/release/5_4/sys/X86_64_GENERIC  x86_64
rustc 1.29.2
   Compiling libc-test v0.1.0 (file:///home/worker/buildbot/dfbsd54-build/build/libc-test)
error: failed to run custom build command for `libc-test v0.1.0 (file:///home/worker/buildbot/dfbsd54-build/build/libc-test)`
process didn't exit successfully: `/home/worker/buildbot/dfbsd54-build/build/target/debug/build/libc-test-59a1aa13c4228b36/build-script-build` (exit code: 101)
--- stdout
-----------------------------------------
cargo:rerun-if-changed=../src/lib.rs
cargo:rerun-if-changed=../src/macros.rs
cargo:rerun-if-changed=../src/dox.rs
cargo:rerun-if-changed=../src/unix/mod.rs
cargo:rerun-if-changed=../src/unix/bsd/mod.rs
cargo:rerun-if-changed=../src/unix/bsd/freebsdlike/mod.rs
cargo:rerun-if-changed=../src/unix/bsd/freebsdlike/dragonfly/mod.rs
OPT_LEVEL = Some("0")
HOST = Some("x86_64-unknown-dragonfly")
CC_x86_64-unknown-dragonfly = None
CC_x86_64_unknown_dragonfly = None
HOST_CC = None
CC = None
CFLAGS_x86_64-unknown-dragonfly = None
CFLAGS_x86_64_unknown_dragonfly = None
HOST_CFLAGS = None
CFLAGS = None
DEBUG = Some("true")
running: "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "-m64" "-Wall" "-Wextra" "-Wall" "-Wextra" "-Werror" "-Wno-unused-parameter" "-Wno-type-limits" "-Wno-address-of-packed-member" "-Wno-deprecated-declarations" "-Wno-deprecated-declarations" "-o" "/home/worker/buildbot/dfbsd54-build/build/target/debug/build/libc-test-9ca12629c0b34c11/out/main.o" "-c" "/home/worker/buildbot/dfbsd54-build/build/target/debug/build/libc-test-9ca12629c0b34c11/out/main.c"
cargo:warning=/home/worker/buildbot/dfbsd54-build/build/target/debug/build/libc-test-9ca12629c0b34c11/out/main.c:65:10: fatal error: sys/ioctl_compat.h: No such file or directory
cargo:warning= #include <sys/ioctl_compat.h>
cargo:warning=          ^~~~~~~~~~~~~~~~~~~~
cargo:warning=compilation terminated.
exit code: 1

--- stderr
thread 'main' panicked at '

Internal error occurred: Command "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "-m64" "-Wall" "-Wextra" "-Wall" "-Wextra" "-Werror" "-Wno-unused-parameter" "-Wno-type-limits" "-Wno-address-of-packed-member" "-Wno-deprecated-declarations" "-Wno-deprecated-declarations" "-o" "/home/worker/buildbot/dfbsd54-build/build/target/debug/build/libc-test-9ca12629c0b34c11/out/main.o" "-c" "/home/worker/buildbot/dfbsd54-build/build/target/debug/build/libc-test-9ca12629c0b34c11/out/main.c" with args "cc" did not execute successfully (status code exit code: 1).

', /home/worker/.cargo/registry/src/github.com-1ecc6299db9ec823/cc-1.0.28/src/lib.rs:2314:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.

program finished with exit code 101
elapsedTime=5.435860
```

```
$ find /usr/include -name "*ioctl*"
/usr/include/netproto/802_11/ieee80211_ioctl.h
/usr/include/dev/raid/mfi/mfi_ioctl.h
/usr/include/dev/video/bktr/ioctl_bt848.h
/usr/include/dev/video/meteor/ioctl_meteor.h
/usr/include/machine/ioctl_meteor.h
/usr/include/machine/ioctl_fd.h
/usr/include/machine/ioctl_bt848.h
/usr/include/sys/ioctl.h
/usr/include/sys/pioctl.h
/usr/include/sys/vnioctl.h
/usr/include/sys/mpt_ioctl.h
/usr/include/sys/mapped_ioctl.h
/usr/include/sys/xdiskioctl.h
/usr/include/bus/u4b/usb_ioctl.h
/usr/include/bus/u4b/ufm_ioctl.h
/usr/include/vfs/autofs/autofs_ioctl.h
/usr/include/vfs/hammer2/hammer2_ioctl.h
/usr/include/vfs/hammer/hammer_ioctl.h
```

```
$ cc --version
cc 8.1 [DragonFly] Release/2018-05-02
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

--- a/libc-test/build.rs
+++ b/libc-test/build.rs
@@ -341,7 +341,6 @@ fn main() {
         cfg.header("mqueue.h");
         cfg.header("ufs/ufs/quota.h");
         cfg.header("pthread_np.h");
-        cfg.header("sys/ioctl_compat.h");
         cfg.header("sys/rtprio.h");
     }
 

